crates/proto/src/rr/rdata/a.rs

```diff
@@ -40,11 +40,51 @@
 //! "10.2.0.52" or "192.0.5.6").
 //! ```
 
-use std::net::Ipv4Addr;
+pub use std::net::Ipv4Addr;
 
 use crate::error::*;
+use crate::rr::{RData, RecordData, RecordType};
 use crate::serialize::binary::*;
 
+impl RecordData for Ipv4Addr {
```

Comment:
From what I've seen, these implementations for `RecordData` for `Ipv4Addr` and `Ipv6Addr` stand out because they're the only impls for types that aren't local wrappers specific to the wrapper type. By contrast, you decided to implement wrappers (through a macro) for all record types that contain a `Name`. Shouldn't we also have `A` and `AAAA` wrappers for these instead of implementing `RecordData` directly for the `std::net::Ipv*Addr` types?

Reply:
This was mainly a decision around backward compatibility. I agree that for consistency with the other types a wrapper type is probably the correct thing. Do you think it's worth considering backward compatibility?

For the A and AAAA types, I'll implement Deref, which maybe is enough?

Reply:
Yeah, `Deref` makes sense.

I usually think adopting the correct design is more important than backwards compatibility, but I don't have a good sense of how impactful this would be to our direct downstream dependencies (or rather, my feeling is that not a lot of downstream users interact with the API at this low level).